qt/Application.h

```diff
@@ -21,14 +21,14 @@
 #include "AddData.h"
 #include "Typedefs.h"
 #include "Utils.h" // std::hash<QString>
+#include "MainWindow.h"
+#include "Prefs.h"
+#include "Session.h"
+#include "TorrentModel.h"
+#include "WatchDir.h"
 
 class AddData;
-class MainWindow;
-class Prefs;
-class Session;
 class Torrent;
-class TorrentModel;
-class WatchDir;
```

Comment:
What is this for?

Reply:
Oh now I see the commit message, did compilation fail?

Reply:
Yeah, unique_ptr::deleter-s for these fields 
https://github.com/transmission/transmission/blob/main/qt/Application.h#L102-L106
couldn't instantiate because of incomplete types.

Reply:
To be fair, you don't have to replace them with includes. Types must be complete at the time of destruction, meaning you can just add a destructor for `Application` and implement it (even if with `= default`) in .cpp file, where full declarations of those classes are available. Since copy and move constructors/operators are already explicitly deleted, the rule of five will be complete.

Reply:
Thanks, Mike, you're right. Done!